addons/account/wizard/account_automatic_entry_wizard.py

```diff
@@ -38,38 +43,87 @@ class AutomaticEntryWizard(models.TransientModel):
-
-    @api.depends('total_amount', 'move_line_ids')
-    def _compute_percentage(self):
-        for record in self:
-            total = (sum(record.move_line_ids.mapped('balance')) or record.total_amount)
-            if total != 0:
-                record.percentage = (record.total_amount / total) * 100
+    @api.depends('chg_period_date_from', 'chg_period_date_to', 'chg_period_time_division', 'chg_period_percentage')
+    def _compute_chg_period_line_ids(self):
+        for wizard in self:
+            one2many_commands = [(5, 0)]
+
+            date_from = wizard.chg_period_date_from
+            date_to = wizard.chg_period_date_to
+            percentage = (wizard.chg_period_percentage or 0.0) / 100.0
+
+            if date_from and date_to and percentage:
+                total_balance = wizard.company_currency_id.round(sum(wizard.move_line_ids.mapped('balance')) * percentage)
+                total_days = 0.0
+                date_distribution = []
+                while date_from <= date_to:
+                    next_month_date = (date_from + relativedelta(months=1)).replace(day=1)
+                    period_date = next_month_date - relativedelta(days=1)
+                    period_days = (period_date - date_from).days
+                    date_distribution.append((period_date, period_days))
+                    date_from = next_month_date
+                    total_days += period_days
+
+                total_reported_balance = 0.0
+                for period_date, nb_days in date_distribution:
+                    if wizard.chg_period_time_division:
+                        period_balance = wizard.company_currency_id.round(total_balance * nb_days / total_days)
+                    else:
+                        period_balance = wizard.company_currency_id.round(total_balance / len(date_distribution))
+                    one2many_commands.append((0, 0, {
+                        'wizard_id': wizard.id,
+                        'date': period_date,
+                        'balance': period_balance,
+                    }))
+                    total_reported_balance += period_balance
+
+                if wizard.chg_period_percentage == 100.0:
+                    total_rounding_error = wizard.company_currency_id.round(total_balance - total_reported_balance)
+                    nber_rounding_steps = int(abs(total_rounding_error / wizard.company_currency_id.rounding))
+                    rounding_error = round(
+                        nber_rounding_steps and total_rounding_error / nber_rounding_steps or 0.0,
+                        wizard.company_currency_id.decimal_places,
+                    )
+                    nber_periods = len(one2many_commands) - 1
+                    if len(one2many_commands) > nber_rounding_steps:
+                        for i in range(nber_rounding_steps):
+                            one2many_commands[nber_periods - i][2]['balance'] += rounding_error
```

Comment:
so I suppose rounding errors are fixed by the last entry. What if it was made by running the wizard multiple times? will it work?

Reply:
by the way: running the wizard multiple times should not try to make the cut off entries for the previous cutoff entries in the original move. I hope you tested that...

Reply:
This task is only an improvement of the previous wizard so this case should already be handled generically... 
When I see
```
    @api.depends('move_line_ids.balance')
    def _compute_from_move_line_ids(self):
        for wizard in self:
            wizard.total_amount = sum(wizard.move_line_ids.mapped('balance'))
            wizard.account_type = 'income' if wizard.total_amount < 0.0 else 'expense'
```
I have the feeling there is an issue there but this should be handled by a fix in previous version.

Reply:
Edit: I'm now using `amount_residual` if the account is a reconcile one but I don't have any way to track there is already some entries generated by a previous adjustment.